runtime/dart_isolate.cc

```diff
@@ -1130,6 +1131,30 @@ Dart_Handle DartIsolate::OnDartLoadLibrary(intptr_t loading_unit_id) {
   return Dart_NewApiError(error_message.c_str());
 }
 
+Dart_Handle DartIsolate::LoadLibraryFromKernel(
+    const std::shared_ptr<const fml::Mapping>& mapping) {
+  if (DartVM::IsRunningPrecompiledCode()) {
+    return Dart_Null();
+  }
+
+  auto* isolate_group_data =
+      static_cast<std::shared_ptr<DartIsolateGroupData>*>(
+          Dart_CurrentIsolateGroupData());
+  // Mapping must be retained until isolate shutdown.
+  (*isolate_group_data)->AddKernelBuffer(mapping);
+
+  auto lib =
+      Dart_LoadLibraryFromKernel(mapping->GetMapping(), mapping->GetSize());
+  if (tonic::CheckAndHandleError(lib)) {
+    return Dart_Null();
```

Comment:
The pattern in the engine is for the Dart code to handle such error conditions without relying on making API calls that would fail to run C++ destructors (such as Dart_ThrowException).  I'm open to arguments about reworking that, but would like to stick to the style used elsewhere in the engine in this patch.